packages/components/src/mobile/link-settings/link-settings-screen.native.js

```diff
@@ -20,17 +20,21 @@ const LinkSettingsScreen = ( props ) => {
 	const { inputValue = url } = route.params || {};
 
 	const onLinkCellPressed = () => {
-		navigation.navigate( 'linkPicker', { inputValue } );
+		if ( props.onLinkCellPressed ) {
+			props.onLinkCellPressed( { navigation } );
```

Comment:
Also, not suggesting a change here, just curious about the scenario where the url was stale. I saw this mentioned in this commit: https://github.com/WordPress/gutenberg/commit/9a806c355d9fe9c9dc346bdb1707cc8914a38f61, and I'm wondering how it came to be stale, since `inputValue` is in the dependencies array of the `useMemo` :thinking: 

Reason I'm curious is that I have noticed some regressions in the link picker (not in your PR) and haven't tracked the root causes down yet. What you encountered might shed light on some possible causes.

Reply:
Unfortunately, I do not have a good, clean answer for you regarding the stale value. It was quite puzzling. With https://github.com/WordPress/gutenberg/commit/9a806c355d9fe9c9dc346bdb1707cc8914a38f61 I attempted to work around the issue by using a different value. 

However, the stale `inputValue` issue became apparent again in a different context later. That discovery—as well as a need to avoid [a React Navigation warning](https://reactnavigation.org/docs/troubleshooting/#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state)—led me to refactor the overall approach in https://github.com/WordPress/gutenberg/commit/d850dd35dc7931b09a835fb4cd316b580ff05d09. Not only did I relocate attribute updating to avoid passing a function in route params, I also [explicitly reset the `inputValue`](https://github.com/WordPress/gutenberg/commit/d850dd35dc7931b09a835fb4cd316b580ff05d09#diff-1925dc0bdda08deb13d65bbf9a16ec62d9905146b6bebf59734048261d999403R85-R86) when a new link destination was selected. 

So, the staleness could be related to my original implementation ignoring React Navigation's warning about non-serializable route params, but I my hunch is that it is more likely related to lexical scope and the parent-child relationship. The callback is defined in a parent render, the child is the one invoking the callback with (originally) route params.

Reply:
> Unfortunately, I do not have a good, clean answer for you regarding the stale value.

Actually, your comment is very helpful, so thanks for the extra context! Regarding the function as a route param or the binding for the passed function, it sounds like the combination of these things could lead to inconsistent state. I'm glad you found a solution that works :+1: .